### PR TITLE
SDK-96 - Added JDK choose option - WIP

### PR DESCRIPTION
**Pull request by tmarzeion**
Feature still needs tests coverage -> Work in progress

### CODE REVIEW

File: src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java [hunk 1]
```diff
@@ -260,6 +290,116 @@ public String promptForPlatformVersion(List<String> versions) {
...
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return result;
+        }
+    }
+
```
Comment (tmarzeion) [inline src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java:394]: @rkorytkowski What do You think about all this File I/O Classes?

File: src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java [hunk 2]
```diff
@@ -260,6 +287,111 @@ public String promptForPlatformVersion(List<String> versions) {
...
+        File openMRS = new File(home, SDKConstants.OPENMRS_SERVER_PATH + "/sdk.properties");
+
+        List<String> result = new ArrayList<String>();
+
+        if(!openMRS.exists()) {
+            return result;
+        }
+        else {
+            try {
+                LineIterator it = FileUtils.lineIterator(openMRS, "UTF-8");
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java:375]: Use new Properties(file); to read properties.

File: src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java [hunk 3]
```diff
@@ -260,6 +287,111 @@ public String promptForPlatformVersion(List<String> versions) {
...
+                if (jdkPath.equals(it.nextLine())) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean isThereJdkUnderPath(String jdkPath) {
+        File jdk = new File(jdkPath + "/bin/java");
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java:353]: Do not hardcode path separators '/'. It won't work on Windows. new File(jdkPath + File.separator + "bin" + File.separator + "java").

File: src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java [hunk 4]
```diff
@@ -260,6 +287,111 @@ public String promptForPlatformVersion(List<String> versions) {
...
+        else {
+            return false;
+        }
+    }
+
+    @Override
+    public List<String> getListOf5RecentJdkPaths() {
+        final int count = 5;
+        String home = System.getProperty("user.home");
+        File openMRS = new File(home, SDKConstants.OPENMRS_SERVER_PATH + "/sdk.properties");
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java:366]: See Server.getServersPathFile() and change to Properties sdkProperties = Server.getServersProperties();

File: src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java [hunk 5]
```diff
@@ -260,6 +287,111 @@ public String promptForPlatformVersion(List<String> versions) {
...
+            return result;
+        }
+        else {
+            try {
+                LineIterator it = FileUtils.lineIterator(openMRS, "UTF-8");
+                for (int j = 0; j < count && it.hasNext(); j++) {
+                    String pathToAdd = it.nextLine();
+                    if (pathToAdd.length() > 0) {
+                        if (isThereJdkUnderPath(pathToAdd)) {
+                            result.add(pathToAdd);
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java:380]: if not correct then remove the entry from sdk.properties.

**Comment by rkorytkowski** [comment]
The repo moved to https://github.com/openmrs/openmrs-sdk, please send pull requests there.

**Pull request closed without merging**